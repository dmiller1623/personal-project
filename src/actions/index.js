export const searchRelated = (searchedRelated) => ({
  type: 'SEARCH_RELATED',
  searchedRelated
}) 