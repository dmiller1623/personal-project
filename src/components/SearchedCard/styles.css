.searched-card {
  display: flex;
  flex-direction: column;
  background-color: #E4E4E1;
  background-image: radial-gradient(at top center, rgba(255, 255, 255, 0.03) 0%, rgba(0, 0, 0, 0.03) 100%), linear-gradient(to top, rgba(255, 255, 255, 0.1) 0%, rgba(143, 152, 157, 0.6) 100%);
  background-blend-mode: normal, multiply;
  height: 24rem;
  margin: 1rem;
  border-radius: 15px;
  padding: .2rem .6rem; }

.card-info {
  width: 100%;
  display: flex;
  flex-direction: row; }

.card-title {
  display: flex;
  justify-content: center; }

.card-description {
  width: 65%;
  display: flex; }

.card-links {
  width: 35%;
  flex-direction: column;
  display: flex; }

.add-button {
  width: 2rem;
  height: 2rem; }

.add-button:hover {
  cursor: pointer; }

h2 {
  color: black;
  font-weight: bold; }

p {
  color: black; }
