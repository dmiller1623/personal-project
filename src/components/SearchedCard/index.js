import React, { Component } from 'react'
import './styles.css'
import addSvg from '../../images/plus.svg'
import ShowMoreText from 'react-show-more-text';
import YouTube from 'react-youtube';
import minusSvg from '../../images/minus-symbol.svg'


// const SearchedCard = (props) => {
//   return (
//     <article className='searched-card'>
//       <div className='card-title'>
//         <div className='button-section'>
//           <img src={addSvg} className='add-button' alt='add button svg' onClick={() => props.addResources(props)}/>
//         </div>
//         <div className='search-heading'>
//           <h2>{props.Type}:</h2>
//           <h2>{props.Name}</h2>
//         </div>
//       </div>
//       <div className='card-info'>
//         <div className='card-description'> 
//           <ShowMoreText>{props.wTeaser}</ShowMoreText>
//         </div>
//         <div className='card-links'>
//         {props.yUrl &&
//           <YouTube
//             className='youtube-video'
//             videoId={props.yUrl.slice(39, 50)}
//             opts={{
//               height: '300',
//               width: '300',
//               autoplay: 1
//             }}
//           />}
//           <a href={props.wUrl}>Wikipedia Link</a>
//         </div>
//       </div>
//     </article>
//   )
// }

// export default SearchedCard

class SearchedCard extends Component {
  constructor(props) {
    super(props)
    this.state = {
      clicked: false
    }
  }

  switchClass = () => {
    const clicked = !this.state.clicked
    this.setState({ clicked })
    this.props.addResources(this.props)
  }

  render() {
    return (
      <article className='searched-card'>
        <div className='card-title'>
          <div className='button-section'>
            <img src={this.state.clicked ? minusSvg : addSvg} className='add-button' alt='add button svg' onClick={this.switchClass}/>
          </div>
          <div className='search-heading'>
            <h2>{this.props.Type}:</h2>
            <h2>{this.props.Name}</h2>
          </div>
        </div>
        <div className='card-info'>
          <div className='card-description'> 
            <ShowMoreText>{this.props.wTeaser}</ShowMoreText>
        </div>
        <div className='card-links'>
          {this.props.yUrl &&
          <YouTube
            className='youtube-video'
            videoId={this.props.yUrl.slice(39, 50)}
            opts={{
            height: '300',
            width: '300',
            autoplay: 1
            }}
          />}
          <a href={this.props.wUrl}>Wikipedia Link</a>
        </div>
      </div>
    </article>
    )
  }
}

export default SearchedCard