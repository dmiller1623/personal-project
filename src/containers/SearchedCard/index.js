import React, { Component } from 'react';
import { connect } from 'react-redux';
import './styles.css';
import addSvg from '../../images/plus.svg';
import ShowMoreText from 'react-show-more-text';
import YouTube from 'react-youtube';
import minusSvg from '../../images/minus-symbol.svg';
import { removeResource } from '../../actions';
import PropTypes from 'prop-types';

export class SearchedCard extends Component {
  constructor(props) {
    super(props);
    this.state = {
      clicked: false
    };
  }

  filterResources = (resource) => {
    const { removeResource, addResources } = this.props;
    const clicked = !this.state.clicked;
    this.setState({ clicked });
    const resourceNames = this.props.additionalResources.map(resource => resource.Name);
    if (resourceNames.includes(resource.Name)) {
      let additionalResources = this.props.additionalResources.filter(name => {
        return name.Name !== resource.Name;
      });
      removeResource(additionalResources);
    } 
    addResources(this.props);
  }

  render() {
    const { Type, Name, wTeaser, yUrl, wUrl } = this.props;
    return (
      <article className='searched-card'>
        <div className='card-title'>
          <div className='button-section'>
            <img src={this.state.clicked ? minusSvg : addSvg} className='add-button' alt='add button svg' onClick={() => this.filterResources(this.props)}/>
          </div>
          <div className='search-heading'>
            <h1>{Type}:</h1>
            <h1 className='searched-card-title'>{Name}</h1>
          </div>
        </div>
        <div className='card-info'>
          <div className='card-description'> 
            <ShowMoreText>{wTeaser}</ShowMoreText>
          </div>
          <div className='card-links'>
            {this.props.yUrl &&
          <YouTube
            className='youtube-video'
            videoId={yUrl.slice(39, 50)}
            opts={{
              height: '300',
              width: '300',
              autoplay: 1
            }}
          />}
            <a href={wUrl}>Wikipedia Link</a>
          </div>
        </div>
      </article>
    );
  }
}

SearchedCard.propTypes = {
  addResources: PropTypes.func,
  additionalResources: PropTypes.array,
  deleteResources: PropTypes.func,
  removeResource: PropTypes.func,
  Name: PropTypes.string,
  Type: PropTypes.string,
  wTeaser: PropTypes.string,
  yUrl: PropTypes.string,
  wUrl: PropTypes.string
};

export const mapStateToProps = (state) => ({
  additionalResources: state.additionalResources
});

export const mapDispatchToProps = (dispatch) => ({
  removeResource: (resource) => dispatch(removeResource(resource))
});

export default connect(mapStateToProps, mapDispatchToProps)(SearchedCard);