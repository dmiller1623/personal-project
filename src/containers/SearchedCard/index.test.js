import React from 'react';
import { shallow, mount } from 'enzyme';
import { mapStateToProps, mapDispatchToProps, SearchedCard } from '../SearchedCard';
import { removeResource } from '../../actions';
import YouTube from 'react-youtube';


describe('SearchedCard', () => {
  let wrapper ;
  it('should match the snapshot', () => {
    wrapper = shallow(<SearchedCard />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should call remove resources when the delete button is clicked and something matches in the array', () => {
    let removeResource = jest.fn();
    let addResources = jest.fn();
    let additionalResources= [{ name: 'jim morrison' }];
    wrapper = shallow(<SearchedCard addResources={addResources} additionalResources={additionalResources} removeResource={removeResource}/>);

    wrapper.find('img').simulate('click');
    expect(removeResource).toHaveBeenCalled();
  });

  it('should call add resources when the add button is clicked and a match is not in the array', () => {
    let removeResource = jest.fn();
    let addResources = jest.fn();
    let additionalResources= [];
    wrapper = shallow(<SearchedCard addResources={addResources} additionalResources={additionalResources} removeResource={removeResource}/>);

    wrapper.find('img').simulate('click');
    expect(addResources).toHaveBeenCalled();
  });

  it('should find a youtube component if there is not a url', () => {
    let additionalResources= [{name: 'jim morrison' }];
    let removeResource = jest.fn();
    let addResources = jest.fn();

    wrapper = mount(<SearchedCard addResources={addResources} additionalResources={additionalResources} removeResource={removeResource}/>);

    expect(wrapper.find(YouTube).length).toEqual(0);

  });

  describe('mapStateToProps', () => {
    it('should return an array of additional resources', () => {
      const store = {
        searchedResources: [],
        additionalResources: []
      };
  
      const expected = {
        additionalResources: []
      };
  
      const mappedProps = mapStateToProps(store);
      expect(mappedProps).toEqual(expected);

    });
  });

  describe('mapDispatchToProps', () => {
    it('should call dispatch when using a function MTDP', () => {
      const mockedDispatch = jest.fn();
      const actionToDispatch = removeResource();
      const mappedProps = mapDispatchToProps(mockedDispatch);

      mappedProps.removeResource();
      expect(mockedDispatch).toHaveBeenCalledWith(actionToDispatch);
    });

  });

});