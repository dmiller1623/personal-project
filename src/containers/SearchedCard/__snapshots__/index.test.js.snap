// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchedCard should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SearchedCard />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="card-title"
>
          <div
                    className="button-section"
          >
                    <img
                              alt="add button svg"
                              className="add-button"
                              onClick={[Function]}
                              src="plus.svg"
                    />
          </div>
          <div
                    className="search-heading"
          >
                    <h1>
                              :
                    </h1>
                    <h1
                              className="searched-card-title"
                    />
          </div>
</div>,
        <div
          className="card-info"
>
          <div
                    className="card-description"
          >
                    <ShowMoreText
                              anchorClass=""
                              less="Show less"
                              lines={3}
                              more="Show more"
                              onClick={undefined}
                    />
          </div>
          <div
                    className="card-links"
          >
                    <a
                              href={undefined}
                    >
                              Wikipedia Link
                    </a>
          </div>
</div>,
      ],
      "className": "searched-card",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="button-section"
>
              <img
                            alt="add button svg"
                            className="add-button"
                            onClick={[Function]}
                            src="plus.svg"
              />
</div>,
            <div
              className="search-heading"
>
              <h1>
                            :
              </h1>
              <h1
                            className="searched-card-title"
              />
</div>,
          ],
          "className": "card-title",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <img
                alt="add button svg"
                className="add-button"
                onClick={[Function]}
                src="plus.svg"
/>,
              "className": "button-section",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "add button svg",
                "className": "add-button",
                "onClick": [Function],
                "src": "plus.svg",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h1>
                  :
</h1>,
                <h1
                  className="searched-card-title"
/>,
              ],
              "className": "search-heading",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    undefined,
                    ":",
                  ],
                },
                "ref": null,
                "rendered": Array [
                  undefined,
                  ":",
                ],
                "type": "h1",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": undefined,
                  "className": "searched-card-title",
                },
                "ref": null,
                "rendered": null,
                "type": "h1",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="card-description"
>
              <ShowMoreText
                            anchorClass=""
                            less="Show less"
                            lines={3}
                            more="Show more"
                            onClick={undefined}
              />
</div>,
            <div
              className="card-links"
>
              <a
                            href={undefined}
              >
                            Wikipedia Link
              </a>
</div>,
          ],
          "className": "card-info",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <ShowMoreText
                anchorClass=""
                less="Show less"
                lines={3}
                more="Show more"
                onClick={undefined}
/>,
              "className": "card-description",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "anchorClass": "",
                "children": undefined,
                "less": "Show less",
                "lines": 3,
                "more": "Show more",
                "onClick": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                undefined,
                <a
                  href={undefined}
>
                  Wikipedia Link
</a>,
              ],
              "className": "card-links",
            },
            "ref": null,
            "rendered": Array [
              undefined,
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Wikipedia Link",
                  "href": undefined,
                },
                "ref": null,
                "rendered": "Wikipedia Link",
                "type": "a",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
    ],
    "type": "article",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="card-title"
>
            <div
                        className="button-section"
            >
                        <img
                                    alt="add button svg"
                                    className="add-button"
                                    onClick={[Function]}
                                    src="plus.svg"
                        />
            </div>
            <div
                        className="search-heading"
            >
                        <h1>
                                    :
                        </h1>
                        <h1
                                    className="searched-card-title"
                        />
            </div>
</div>,
          <div
            className="card-info"
>
            <div
                        className="card-description"
            >
                        <ShowMoreText
                                    anchorClass=""
                                    less="Show less"
                                    lines={3}
                                    more="Show more"
                                    onClick={undefined}
                        />
            </div>
            <div
                        className="card-links"
            >
                        <a
                                    href={undefined}
                        >
                                    Wikipedia Link
                        </a>
            </div>
</div>,
        ],
        "className": "searched-card",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="button-section"
>
                <img
                                alt="add button svg"
                                className="add-button"
                                onClick={[Function]}
                                src="plus.svg"
                />
</div>,
              <div
                className="search-heading"
>
                <h1>
                                :
                </h1>
                <h1
                                className="searched-card-title"
                />
</div>,
            ],
            "className": "card-title",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <img
                  alt="add button svg"
                  className="add-button"
                  onClick={[Function]}
                  src="plus.svg"
/>,
                "className": "button-section",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "add button svg",
                  "className": "add-button",
                  "onClick": [Function],
                  "src": "plus.svg",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <h1>
                    :
</h1>,
                  <h1
                    className="searched-card-title"
/>,
                ],
                "className": "search-heading",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      undefined,
                      ":",
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    undefined,
                    ":",
                  ],
                  "type": "h1",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": undefined,
                    "className": "searched-card-title",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "h1",
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="card-description"
>
                <ShowMoreText
                                anchorClass=""
                                less="Show less"
                                lines={3}
                                more="Show more"
                                onClick={undefined}
                />
</div>,
              <div
                className="card-links"
>
                <a
                                href={undefined}
                >
                                Wikipedia Link
                </a>
</div>,
            ],
            "className": "card-info",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <ShowMoreText
                  anchorClass=""
                  less="Show less"
                  lines={3}
                  more="Show more"
                  onClick={undefined}
/>,
                "className": "card-description",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "anchorClass": "",
                  "children": undefined,
                  "less": "Show less",
                  "lines": 3,
                  "more": "Show more",
                  "onClick": undefined,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  undefined,
                  <a
                    href={undefined}
>
                    Wikipedia Link
</a>,
                ],
                "className": "card-links",
              },
              "ref": null,
              "rendered": Array [
                undefined,
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Wikipedia Link",
                    "href": undefined,
                  },
                  "ref": null,
                  "rendered": "Wikipedia Link",
                  "type": "a",
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "article",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
